### PR TITLE
Kevinsolorio/rep 15 viewing claiming pickup tasks create simple pickup tasks

### PR DESCRIPTION
**Pull request by kevin3656**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Added available pickups page where listings of pickups are displayed based on the calendar date chosen. The calendar gets the current date and updates accordingly. The pressable listings redirect you to a pickup details page that is very barebones right now but it has a claim pickup button.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 
IMG_1320
IMG_1319
IMG_1321

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
the main files are available-pickups/index.tsx and pickup-details./_layout.tsx. You can change the dates of the mock listings in available-pickups and they should show up on the corresponding date when you press it on the calendar, should also be listed by time of pickups. 

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
needs styling, the routing also will probably change since right now i just have app/index redirecting to available-pickups

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @insert pl github username here

### CODE REVIEW

**Review by torinnayak1 — APPROVED** [review]
In line styling but rest is goated, please let me suck your dick king

**Review by jeannineruiz — APPROVED** [review]
LGTM

File: src/app/available-pick-ups/index.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        const day = d.toLocaleDateString(undefined, { day: '2-digit' });
+        const dow = d.toLocaleDateString(undefined, { weekday: 'short' }); // Mon
+
+        return (
+          <Pressable
+            key={iso}
+            onPress={() => onSelect(iso)}
+            style={({ pressed }) => ({
+              width: 56,
+              height: 72,
```
Comment (rachaelch3n) [inline src/app/available-pick-ups/index.tsx]: can we make a style component for this? you can add conditional styles like `style={[styles.dateCard, isSelected && styles.selectedCard]}
`

File: src/app/available-pick-ups/index.tsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+              justifyContent: 'center',
+              borderWidth: 1,
+              borderColor: isSelected ? '#000' : '#d1d5db',
+              backgroundColor: isSelected ? '#000' : '#f3f4f6',
+              marginRight: 8, // spacing between chips
+              opacity: pressed ? 0.7 : 1,
+            })}
+          >
+            <Text
+              style={{ fontSize: 10, color: isSelected ? 'white' : 'black' }}
```
Comment (rachaelch3n) [inline src/app/available-pick-ups/index.tsx]: style component?

File: src/app/available-pick-ups/index.tsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            >
+              {month}
+            </Text>
+            <Text
+              style={{
+                fontSize: 16,
+                fontWeight: '700',
```
Comment (rachaelch3n) [inline src/app/available-pick-ups/index.tsx]: also marking this for style component

File: src/app/available-pick-ups/index.tsx [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+                color: isSelected ? 'white' : 'black',
+              }}
+            >
+              {day}
+            </Text>
+            <Text
+              style={{ fontSize: 10, color: isSelected ? 'white' : 'black' }}
```
Comment (rachaelch3n) [inline src/app/available-pick-ups/index.tsx]: + this

File: src/app/available-pick-ups/index.tsx [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+            >
+              {dow}
+            </Text>
+            {isToday && (
+              <Text
+                style={{
+                  marginTop: 2,
+                  fontSize: 9,
```
Comment (rachaelch3n) [inline src/app/available-pick-ups/index.tsx]: + this. just add all the components to your already made styles.ts file

File: src/app/available-pick-ups/index.tsx [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+      MOCK_PICKUPS.filter(p => p.slot_start_time.slice(0, 10) === selectedISO),
+    [selectedISO],
+  );
+
+  return (
+    <FlatList
+      data={filtered}
+      keyExtractor={item => String(item.id)}
+      contentContainerStyle={{
+        paddingHorizontal: 16,
```
Comment (rachaelch3n) [inline src/app/available-pick-ups/index.tsx]: style component again

File: src/app/available-pick-ups/index.tsx [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+            days={days}
+            todayISO={todayISO}
+            selectedISO={selectedISO}
+            onSelect={setSelectedISO}
+          />
+          <Text
+            style={{
+              fontSize: 18,
+              fontWeight: '700',
+              marginTop: 40,
```
Comment (rachaelch3n) [inline src/app/available-pick-ups/index.tsx]: style component

File: src/app/available-pick-ups/index.tsx [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+      }
+      ItemSeparatorComponent={() => <View style={{ height: 12 }} />}
+      renderItem={({ item }) => (
+        <Pressable
+          onPress={() => router.push(`/pickup-details/${item.id}`)}
+          style={({ pressed }) => ({
+            padding: 16,
+            borderWidth: 1,
+            borderColor: '#d1d5db',
+            borderRadius: 16,
```
Comment (rachaelch3n) [inline src/app/available-pick-ups/index.tsx:159]: here too

File: src/app/available-pick-ups/index.tsx [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+            backgroundColor: '#eee',
+            opacity: pressed ? 0.6 : 1,
+          })}
+        >
+          <Text style={{ fontWeight: '700', marginBottom: 6 }}>
```
Comment (rachaelch3n) [inline src/app/available-pick-ups/index.tsx:164]: marking here too

File: src/app/available-pick-ups/index.tsx [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+}
+
+export default function AvailablePickupsPage() {
+  const todayISO = new Date().toISOString().slice(0, 10);
+  const [selectedISO, setSelectedISO] = React.useState(todayISO);
+  const days = React.useMemo(() => {
+    const out: string[] = [];
+    const start = new Date();
+    start.setDate(start.getDate() - 1);
+    for (let i = 0; i < 5; i++) {
```
Comment (rachaelch3n) [inline src/app/available-pick-ups/index.tsx]: nice! but since we now only want to render for today and tomorrow, we can change this to 2 instead of 5

File: src/app/pickup-details/_layout.tsx [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <Stack
+      screenOptions={{
+        headerTitle: 'Pickup Details',
+        headerRight: () => (
+          <Pressable
+            onPress={() =>
+              Alert.alert('Claim Pickup', 'Pretend we claimed it 🤓🍻')
+            }
+            style={{ paddingHorizontal: 8, paddingVertical: 6 }}
```
Comment (rachaelch3n) [inline src/app/pickup-details/_layout.tsx:15]: style component

File: src/app/pickup-details/_layout.tsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+          >
+            <Text style={{ fontWeight: '600' }}>Claim Pickup</Text>
```
Comment (rachaelch3n) [inline src/app/pickup-details/_layout.tsx]: style component (even though its just fontweight rn, we might add more later)

File: src/app/available-pick-ups/index.tsx [hunk 13]
```diff
@@ -0,0 +1,202 @@
...
+  {
+    id: 4,
+    slot_start_time: '2025-10-13T15:00:00',
+    slot_end_time: '2025-10-13T18:00:00',
+    pickup_location: 'Kingman Hall',
+  },
+];
+
+// datetime helpers
+const fmtTimeRange = (startISO: string, endISO: string) => {
```
Comment (rachaelch3n) [inline src/app/available-pick-ups/index.tsx]: i'm thinking maybe we put this function into utils (pathway would be ../utils/dateHelpers.ts) for instance, since you also define another function late (fmtTime) in pickup-details/[id].tsx that also formats the date, just doesn't create the range. 

you can do this: `// src/utils/dateHelpers.ts

/** Format a single ISO time to 12-hour format (e.g. "9:00 am") */
export const fmtTime = (iso: string): string => {
  return new Date(iso)
    .toLocaleTimeString(undefined, { hour: 'numeric', minute: '2-digit' })
    .toLowerCase();
};

/** Format two ISO times into a range (e.g. "9:00 am - 10:30 am") */
export const fmtTimeRange = (startISO: string, endISO: string): string => {
  return `${fmtTime(startISO)} - ${fmtTime(endISO)}`;
};
`

File: src/app/pickup-details/[id].tsx [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+  return (
+    <View
+      style={{
+        borderWidth: 1,
+        borderColor: '#d1d5db',
+        borderRadius: 12,
+        padding: 16,
+        marginBottom: 12,
+        backgroundColor: '#f7f7f7',
+      }}
```
Comment (rachaelch3n) [inline src/app/pickup-details/[id].tsx]: style component for readability

File: src/app/pickup-details/[id].tsx [hunk 15]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { Text, View } from 'react-native';
+import { useLocalSearchParams } from 'expo-router';
+import { MOCK_PICKUPS } from '../available-pick-ups'; // re-use your mock data
+
+function fmtTime(iso: string) {
+  return new Date(iso)
+    .toLocaleTimeString(undefined, { hour: 'numeric', minute: '2-digit' })
+    .toLowerCase();
```
Comment (rachaelch3n) [inline src/app/pickup-details/[id].tsx]: noting that we can add this to a shared dateHelpers.ts file

File: src/app/pickup-details/[id].tsx [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+      {children}
+    </View>
+  );
+}
+
+export default function PickupDetails() {
+  const { id } = useLocalSearchParams<{ id: string }>();
+  const pickup = MOCK_PICKUPS.find(p => String(p.id) === String(id));
+
+  if (!pickup) {
```
Comment (rachaelch3n) [inline src/app/pickup-details/[id].tsx:27]: nice error handling. also inline styling

**Review by rachaelch3n — CHANGES_REQUESTED** [review]
